### ISSUE TITLE
failed to get IPs for node

### ISSUE THREAD

**Issue by gengwg** [kubernetes-sigs/kind#1149]
```
$ kind create cluster --loglevel debug
WARNING: --loglevel is deprecated, please switch to -v and -q!
Creating cluster "kind" ...
DEBUG: docker/images.go:58] Image: kindest/node:v1.16.3 present locally
 ✓ Ensuring node image (kindest/node:v1.16.3) 🖼
 ✓ Preparing nodes 📦
 ✗ Writing configuration 📜
ERROR: failed to create cluster: failed to get IPs for node: kind-control-plane: file should only be one line, got 2 lines

Stack Trace:
sigs.k8s.io/kind/pkg/errors.Errorf
	/private/tmp/kind-20191204-84812-92plwg/kind-0.6.1/pkg/errors/errors.go:33
sigs.k8s.io/kind/pkg/internal/cluster/providers/docker.(*node).IP
	/private/tmp/kind-20191204-84812-92plwg/kind-0.6.1/pkg/internal/cluster/providers/docker/node.go:63
sigs.k8s.io/kind/pkg/cluster/nodeutils.GetControlPlaneEndpoint
	/private/tmp/kind-20191204-84812-92plwg/kind-0.6.1/pkg/cluster/nodeutils/util.go:44
sigs.k8s.io/kind/pkg/internal/cluster/create/actions/config.(*Action).Execute
	/private/tmp/kind-20191204-84812-92plwg/kind-0.6.1/pkg/internal/cluster/create/actions/config/config.go:56
sigs.k8s.io/kind/pkg/internal/cluster/create.Cluster
	/private/tmp/kind-20191204-84812-92plwg/kind-0.6.1/pkg/internal/cluster/create/create.go:136
sigs.k8s.io/kind/pkg/cluster.(*Provider).Create
	/private/tmp/kind-20191204-84812-92plwg/kind-0.6.1/pkg/cluster/provider.go:100
sigs.k8s.io/kind/pkg/cmd/kind/create/cluster.runE
	/private/tmp/kind-20191204-84812-92plwg/kind-0.6.1/pkg/cmd/kind/create/cluster/createcluster.go:86
sigs.k8s.io/kind/pkg/cmd/kind/create/cluster.NewCommand.func1
	/private/tmp/kind-20191204-84812-92plwg/kind-0.6.1/pkg/cmd/kind/create/cluster/createcluster.go:52
github.com/spf13/cobra.(*Command).execute
	/private/tmp/kind-20191204-84812-92plwg/kind-0.6.1/.brew_home/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826
github.com/spf13/cobra.(*Command).ExecuteC
	/private/tmp/kind-20191204-84812-92plwg/kind-0.6.1/.brew_home/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914
github.com/spf13/cobra.(*Command).Execute
	/private/tmp/kind-20191204-84812-92plwg/kind-0.6.1/.brew_home/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
sigs.k8s.io/kind/cmd/kind/app.Run
	/private/tmp/kind-20191204-84812-92plwg/kind-0.6.1/cmd/kind/app/main.go:53
sigs.k8s.io/kind/cmd/kind/app.Main
	/private/tmp/kind-20191204-84812-92plwg/kind-0.6.1/cmd/kind/app/main.go:35
main.main
	/private/tmp/kind-20191204-84812-92plwg/kind-0.6.1/main.go:25
runtime.main
	/usr/local/Cellar/go/1.13.4/libexec/src/runtime/proc.go:203
runtime.goexit
	/usr/local/Cellar/go/1.13.4/libexec/src/runtime/asm_amd64.s:1357
```

I'm using macbook. Docker version 19.03.5, build 633a0ea

**Comment by BenTheElder** [comment]
whoah, can you share your docker info?

also, it should have included the command output in the full output of that command?

**Comment by BenTheElder** [comment]
er `docker info` that is?

have you configured docker in any non-default ways?

**Comment by gengwg** [comment]
@BenTheElder thank you very much for the prompt response. My coworker (what a genius!) helped me found out the root cause. You are right, my docker is actually an alias. This is a new Mac. I copied my bash alias file from a Linux machine, where it needs sudo to run docker. On Mac, one don't need sudo to run docker command.

```
alias docker='sudo docker'
```

I started a new terminal (to remove the alias), and remove the `.docker/` dir in my home dir (_this is also required_). It is working now.  

I should have debugged why it starts asking me password when I run docker command at the beginning, but thought it's new installation of new version of docker, and didn't pay too much attention..
